cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/UpgradeHttpHandler.java

```diff
@@ -40,17 +40,15 @@
 public class UpgradeHttpHandler extends AbstractHttpHandler {
 
   private static final Gson GSON = new Gson();
+  private Map<String, Boolean> upgradeStatus;
 
-  private final AppVersionUpgradeService appVersionUpgradeService;
-
-  @Inject
-  UpgradeHttpHandler(AppVersionUpgradeService appVersionUpgradeService) {
-    this.appVersionUpgradeService = appVersionUpgradeService;
+  UpgradeHttpHandler() {
+    this.upgradeStatus = new HashMap<>();
   }
 
   @GET
   @Path("/status")
   public void getUpgradeStatus(HttpRequest request, HttpResponder responder) throws Exception {
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(appVersionUpgradeService.getUpgradeStatus()));
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(upgradeStatus));
```

Comment:
Instead of returning empty status. Can we return the status of Metadata upgrade?